test/e2e/operator_expectations.go

```diff
@@ -121,3 +126,48 @@ func ExpectAllMachinesLinkedToANode() error {
 	}
 	return nil
 }
+
+func ExpectReconcileControllersDeployment() error {
+	key := types.NamespacedName{
+		Namespace: namespace,
+		Name:      "clusterapi-manager-controllers",
+	}
+	d := &kappsapi.Deployment{}
+
+	glog.Info("Get deployment")
+	err := wait.PollImmediate(1*time.Second, waitShort, func() (bool, error) {
+		if err := F.Client.Get(context.TODO(), key, d); err != nil {
+			glog.Errorf("error querying api for Deployment object: %v, retrying...", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	glog.Info("Delete deployment")
+	err = wait.PollImmediate(1*time.Second, waitShort, func() (bool, error) {
+		if err := F.Client.Delete(context.TODO(), d); err != nil {
+			glog.Errorf("error querying api for Deployment object: %v, retrying...", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	glog.Info("Verify deployment is recreated")
+	err = wait.PollImmediate(1*time.Second, waitLong, func() (bool, error) {
+		if err := F.Client.Get(context.TODO(), key, d); err != nil {
+			glog.Errorf("error querying api for Deployment object: %v, retrying...", err)
+			return false, nil
+		}
```

Comment:
You should check if the deletion timestamp is zero as well. In case it takes longer to delete the object. Otherwise, the `Get` can actually return to-be deleted object.